Dockerfile

```
@@ -1,29 +1,40 @@
-FROM python:3.6-slim
-
-SHELL ["/bin/bash", "-c"]
-
+FROM python:3.6-slim as python_builder
 RUN apt-get update -qq && \
   apt-get install -y --no-install-recommends \
-  build-essential && \
-  apt-get clean && \
-  rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* && \
-  mkdir /app
-
-WORKDIR /app
-
-# Copy as early as possible so we can cache ...
-COPY requirements.txt .
-
-RUN pip install -r requirements.txt --no-cache-dir
+  build-essential \
+  curl
+
+# install poetry
+# keep this in sync with the version in pyproject.toml and Dockerfile
+ENV POETRY_VERSION 1.0.3
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+ENV PATH "/root/.poetry/bin:/opt/venv/bin:${PATH}"
+
+# install dependencies
+COPY . /app/
+RUN python -m venv /opt/venv && \
+  . /opt/venv/bin/activate && \
+  pip install --no-cache-dir -U pip && \
+  cd /app && \
+  poetry install --no-dev --no-interaction
+
+# start a new build stage
+FROM python:3.6-slim
 
-COPY . .
+# copy everything from /opt
+COPY --from=python_builder /opt/venv /opt/venv
+COPY --from=python_builder /app /app
+ENV PATH="/opt/venv/bin:$PATH"
 
-RUN pip install -e . --no-cache-dir
+# update permissions & change user
+RUN chgrp -R 0 /app && chmod -R g=u /app
+USER 1001
 
-VOLUME ["/app/actions"]
+# change shell
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
+# create a mount point for custom actions and the entry point
+WORKDIR /app/actions
```

with `actions.py` in `/app/actions` `start --actions actions.actions` gonna break otherwise  
```suggestion
WORKDIR /app
```